dace/codegen/targets/cpp.py

```diff
@@ -216,41 +216,101 @@ def memlet_copy_to_absolute_strides(dispatcher,
     return copy_shape, src_strides, dst_strides, src_expr, dst_expr
 
 
-def ptr(name: str, desc: data.Data) -> str:
+def ptr(name: str,
+        desc: data.Data = None,
+        subset_info: Union[subsets.Subset, int] = None,
+        sdfg: dace.SDFG = None,
+        is_write: bool = None,
+        dispatcher=None,
+        ancestor: int = None,
+        is_array_interface: bool = False,
+        interface_id: Union[int, List[int]] = None) -> str:
```

Comment:
No, because one can also have array interfaces that don't have an interface id appended. To be honest I don't really see a good way to save arguments here. 

It' s true that subset_info is just used for HBM, but I would like to keep this name because from the "logic" behind ptr, subset_info is just an optional parameter that contains any info on what will be accessed by this variable (can also be a whole subset not just the first index; and it's frequently also passed for arrays that don't use HBM, though it does not change anything then at the moment)

Reply:
Can you see a potential use case outside of HBM for `subset_info`?

Reply:
Not really ;) Ok I will rename it.